src/core/ReactCompositeComponent.js

```diff
@@ -396,11 +371,32 @@ var ReactCompositeComponentMixin = assign({},
    * @protected
    */
   setState: function(partialState, callback) {
+    var compositeLifeCycleState = this._compositeLifeCycleState;
+    invariant(
+      ReactCurrentOwner.current == null,
+      'setState(...): Cannot update during an existing state transition ' +
+      '(such as within `render`). Render methods should be a pure function ' +
+      'of props and state.'
+    );
+    invariant(
+      compositeLifeCycleState !== CompositeLifeCycle.UNMOUNTING,
+      'setState(...): Cannot call setState() on an unmounting component.'
+    );
     // Merge with `_pendingState` if it exists, otherwise with existing state.
-    this.replaceState(
```

Comment:
Because I aim to deprecate it and make a separate queue for all this stuff. Also, it doesn't make sense to have an error message called replaceState when there is no replaceState method on React.Component.